### PR TITLE
feat: keystone authentication against dex

### PR DESCRIPTION
**Pull request by cardoe**

### CODE REVIEW

File: components/dex/values.yaml [hunk 1]
```diff
@@ -49,6 +49,11 @@ config:
       name: "Undercloud ArgoCD"
       redirectURIs:
         - "https://argocd.$DNS_ZONE/auth/callback"
+    - id: keystone
+      secretEnv: KEYSTONE_SSO_CLIENT_SECRET
+      name: "OpenStack Keystone"
+      redirectURIs:
+        - "https://keystone-api.$DNS_ZONE/redirect_uri"
```
Comment (skrobul) [inline components/dex/values.yaml:56]: `keystone-api` service does not listen on default port 443, so this should either be

`https://keystone-api.$DNS_ZONE:5000/redirect_uri`
or
`https://keystone.$DNS_ZONE/redirect_uri`

judging by the Apache configs below it should be the latter

**Comment by skrobul** [comment]
Depends on https://github.com/RSS-Engineering/undercloud-deploy/pull/140

Closes PUC-498

File: components/dex/values.yaml (hunk 1 above)
Comment (cardoe) [reply components/dex/values.yaml]: Yes. you're right.

File: components/keystone/aio-values.yaml [hunk 2]
```diff
@@ -33,6 +33,48 @@ bootstrap:
...
+                    "type": "REMOTE_USER"
+                },
+                {
+                    "type": "HTTP_OIDC_GROUPS"
+                }
+            ]
+        }
+    ]
+    EOF
+    openstack identity provider show sso 2>/dev/null || openstack identity provider create --remote-id https://dex.dev.undercloud.rackspace.net sso
```
Comment (cardoe) [inline components/keystone/aio-values.yaml:65]: TODO: fix this.

**Review by cardoe — COMMENTED** [review]
I can't +1 this cause I started it.

**Review by skrobul — APPROVED** [review]

**Pull request merged**